extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/PythonTableDataService.java

```diff
@@ -863,4 +929,35 @@ public long size() {
     private static class Subscription {
         SafeCloseable cancellationCallback;
     }
+
+    // this tool is used to simplify backend asynchronous RPC patterns for otherwise synchronous operations
+    private static class AsyncState<T> {
+        private T result;
+        private RuntimeException error;
+
+        public synchronized void setResult(@NotNull final T result) {
+            this.result = result;
```

Comment:
I wondered if we should do any kind of state checking to avoid double-setting result/error, but decided it was unnecessary.

Reply:
An assertion could be useful, both that it was not yet set and that the result is actually not null. I'll add them; it's a good point.